pypdf/_writer.py

```diff
@@ -2110,22 +2115,76 @@ def remove_annotations(
                 Examples are: "/Link", "/FileAttachment", "/Sound",
                 "/Movie", "/Screen", ...
                 If you want to remove all annotations, use subtypes=None.
-        """
+            annotation_filter_function: Function that takes three arguments,
+                DictionaryObject, ArrayObject and DictionaryObject, and decides whether to remove them. For example:
+                    def is_google_link(
+                        page: DictionaryObject,
+                        annotation: ArrayObject,
+                        obj: DictionaryObject
```

Comment:
Maybe you can add a little bit more documentation:

* Why is the annotation an ArrayObject?
* What does `obj` represent?

Reply:
I don't actually properly understand these objects. I just kept the original names and annotations, see [this part of the old code](https://github.com/MrTomRod/pypdf/blob/914c43eed09d9153322ccbc2de8f4f16d4b21ba3/pypdf/_writer.py#L1930) where annotations were filtered.

In short, I don't know what to write myself, sorry.